# Case Study on ANOVA

XYZ Company has offices in four different zones. The company wishes to investigate the following :

- The mean sales generated by each zone.
- Total sales generated by all the zones for each month.
- Check whether all the zones generate the same amount of sales.

Help the company to carry out their study with the help of data provided.

In [1]:
#importing pandas,numply,matplotlib.pyplot,seaborn libaries
import pandas as pd
import numpy as np

In [2]:
#Reading the dataset to the python environment
data=pd.read_csv('Sales_data_zone_wise.csv')

In [3]:
#displaying  top 5  rows of the dataset
data.head()

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [4]:
#To know data types of each columns and checking for null values in the columns
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Month     29 non-null     object
 1   Zone - A  29 non-null     int64 
 2   Zone - B  29 non-null     int64 
 3   Zone - C  29 non-null     int64 
 4   Zone - D  29 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 1.3+ KB


<div class="alert alert-block alert-info"><b>Note:</b><br>  
    
- It appears that we do not have empty/NaN values in the dataset,as the number of Non-Null counts for each column is equal to 29, which is exactly the number of rows in the dataset.<br>
- We also note that our numeric data are detected as integer values.
</div>

In [5]:
#Check for the null values present in each column of  dataset using 'isnull().sum()' function
data.isnull().sum()

Month       0
Zone - A    0
Zone - B    0
Zone - C    0
Zone - D    0
dtype: int64

**Note-**

No  <mark>Null values <mark>


In [6]:
# to know  statistical summary of integer datatype columns 
data.describe()

Zone - A      Zone - B      Zone - C      Zone - D
count  2.900000e+01  2.900000e+01  2.900000e+01  2.900000e+01
mean   1.540493e+06  1.755560e+06  1.772871e+06  1.842927e+06
std    2.619401e+05  1.683899e+05  3.331937e+05  3.750165e+05
min    1.128185e+06  1.527574e+06  1.237722e+06  1.234311e+06
25%    1.305972e+06  1.606010e+06  1.523308e+06  1.520406e+06
50%    1.534390e+06  1.740365e+06  1.767047e+06  1.854412e+06
75%    1.820196e+06  1.875658e+06  2.098463e+06  2.180416e+06
max    2.004480e+06  2.091194e+06  2.290580e+06  2.364132e+06

## The company wishes to investigate  :
### 1. The mean sales generated by each zone.

In [7]:
round(data.mean(),2)

Zone - A    1540493.14
Zone - B    1755559.59
Zone - C    1772871.03
Zone - D    1842926.76
dtype: float64

<div class="alert alert-block alert-info"><b>Note:</b><br>  
    
- mean sales generated by  Zone-D is higher.<br>
- mean sales generated by  Zone-A is lower.
</div>

## The company wishes to investigate  :
### 2.Total sales generated by all the zones for each month.

In [11]:
data['Total Sales']=data.iloc[:,0:5].sum(axis=1)
data

Month  Zone - A  Zone - B  Zone - C  Zone - D  Total Sales
0    Month - 1   1483525   1748451   1523308   2267260      7022544
1    Month - 2   1238428   1707421   2212113   1994341      7152303
2    Month - 3   1860771   2091194   1282374   1241600      6475939
3    Month - 4   1871571   1759617   2290580   2252681      8174449
4    Month - 5   1244922   1606010   1818334   1326062      5995328
5    Month - 6   1534390   1573128   1751825   2292044      7151387
6    Month - 7   1820196   1992031   1786826   1688055      7287108
7    Month - 8   1625696   1665534   2161754   2363315      7816299
8    Month - 9   1652644   1873402   1755290   1422059      6703395
9   Month - 10   1852450   1913059   1754314   1608387      7128210
10  Month - 11   2004480   1740365   1317869   1970069      7032783
11  Month - 12   1537539   1875658   1237722   1460165      6111084
12  Month - 13   1254939   1588473   1348629   1733383      5925424
13  Month - 14   1821799   1844081   1617376   1872259      7155515
14  Month - 15   1128185   1804613   1767047   1234311      5934156
15  Month - 16   1491534   1542053   1882159   1590913      6506659
16  Month - 17   1182110   1552388   2234469   2180416      7149383
17  Month - 18   1770324   1699213   1249821   2364132      7083490
18  Month - 19   1181635   1959268   2009168   1821882      6971953
19  Month - 20   1305972   1559670   2107949   2151008      7124599
20  Month - 21   1570152   2013615   1540016   2265814      7389597
21  Month - 22   1481619   1527574   2255729   2295079      7560001
22  Month - 23   1887694   1992155   1348387   1459683      6687919
23  Month - 24   1880820   1752873   2098463   2052591      7784747
24  Month - 25   1256333   1622671   1521792   1695122      6095918
25  Month - 26   1422853   1715465   1853636   1520406      6512360
26  Month - 27   1384426   1983163   1611169   1289160      6267918
27  Month - 28   1616640   1547991   2128022   2178267      7470920
28  Month - 29   1310654   1660092   1947119   1854412      6772277

<div class="alert alert-block alert-info"><b>Find:</b><br>  
    
- Highest Total Sales was  generated in month-4 .<br>
- Lowest  Total Sales was  generated in month-13.
</div>

## The company wishes to investigate  :
### 3.Check whether all the zones generate the same amount of sales.

**One-way ANOVA Test** is used to check whether all the zones generate the same amount of sales <br>
Here one independent variable/factor (sales generated in each zones ) affecting a dependent variable (Total sales)<br>
so One-way ANOVA Test is uesed

>Set the hypothesis 

Null Hypothesis (H0):  all the zones generate the same amount of sales

Alternate Hypothesis (H1): all the zones does not generate the same amount of sales.


>Set the significance level 

Significance level,α= 0.05


>One-way ANOVA Test.

In [9]:
from scipy import stats

F, p = stats.f_oneway(data['Zone - A'], data['Zone - B'], data['Zone - C'],data['Zone - D'])
print('F test statistic:',round(F,4),'p-value:',round(p,4))

F test statistic: 5.6721 p-value: 0.0012


In [10]:
if p >= 0.05:
    print('Accept the Null Hypothesis')
else :
    print ('Reject the  Null Hypothesis')

Reject the  Null Hypothesis


>Making a Decision 

<div class="alert alert-block alert-info"><b>Find:</b><br>  
    
From the One-way ANOVA Test performed,<br>  
p-value: 0.0012<br>  
p-value< 0.05<br>  

So we <code style="background:yellow;color:black">reject the null hypothesis H0.</code><br> 

i.e , <code style="background:yellow;color:black"> all the zones does not generate the same amount of sales.</code><br>
                        
</div>

Submitted by-

`Kevin M
 DSA Norka B3 2021`

---